### DISCUSSION TITLE
F1 API Usage

### DISCUSSION THREAD

**Discussion by joearkay** [slowlydev/f1-dash#117] [General]
Firstly, great project. Love what you have done. 

I'm interested in how the project as well as the hosted-version consume F1 data. I've traced back to [this line](https://github.com/slowlydev/f1-dash/blob/c4b3cba4532217495ef58c0dc9ebff2cfbaf756a/data-saver/src/socket.rs#L11C28-L11C65) which points towards the signalr API wss://livetiming.formula1.com/signalr

Is there some kind of authentication required for this API? I am surprised it can be accessed free-of-charge. How is the hosted web app (f1-dash.com) consuming this information?

Thanks :)

**Comment by joearkay** [comment]
I have perhaps just answered my own question. It seems this endpoint is 'open'?
https://dweik.xyz/post/f1-signalr-endpoint/

**Reply by slowlydev** [reply]
Yes it's "just open", I saw some people send their f1 accounts tokens aswell but it's not needed for the telemetry data, no idea why. In this issue #58 there's a bit more about the topic.

**Comment by slowlydev** [comment]
> Firstly, great project. Love what you have done.

Thank you I appreciate it!
 
> I'm interested in how the project as well as the hosted-version consume F1 data. I've traced back to [this line](https://github.com/slowlydev/f1-dash/blob/c4b3cba4532217495ef58c0dc9ebff2cfbaf756a/data-saver/src/socket.rs#L11C28-L11C65) which points towards the signalr API wss://livetiming.formula1.com/signalr

Yes `wss://livetiming.formula1.com/signalr` is the api for the live timing. `data-saver` is a small old utility to connect to f1 and print out the messages.

The "real" connection used on the hosted is found in `live-backend`.

> Is there some kind of authentication required for this API? I am surprised it can be accessed free-of-charge. How is the hosted web app (f1-dash.com) consuming this information?

`live-backend` does the whole connecting to the signalr web socket. every json we receive we transform into camelCase because I don't like PascalCase then there's one variable that keeps being updated all the updates we receive also gets forwarded to every open SEE connection.

`live-backend` also stores a snapshot every second of the one variable and stores it in a db for analytics purposes (frontend does not show it yet).   

`dash` connects to SSE endpoint and keeps the UI updated, also does fancy buffer stuff to allow for setting delays.

**Comment by joearkay** [comment]
Thank you for the feedback. I'll keep you updated with my usage and will take a further dig at the code this week to get a better understanding. buymeacoffee inbound ☕ 😄
